src/adapters/borges_ets_adapter.erl

```diff
@@ -26,10 +24,17 @@ fetch(Key, #{storage_adapter_config := StorageAdapterConfig}) ->
     TableName = maps:get(name, StorageAdapterConfig),
     case ets:lookup(TableName, Key) of
         [{_, Data}] -> {ok, Data};
-        _ -> {ok, []}
+        _ -> {ok, not_found}
     end.
 
 remove(Key, #{storage_adapter_config := StorageAdapterConfig}) ->
     TableName = maps:get(name, StorageAdapterConfig),
     ets:delete(TableName, Key),
     ok.
+
+keys(#{storage_adapter_config := StorageAdapterConfig}) ->
+    TableName = maps:get(name, StorageAdapterConfig),
+    Keys =
+        lists:flatten(
```

Comment:
Can Keys be strings (lists)? In this case they risk to be flattened into one string..

Reply:
Yes they can, nice catch.